scipy/optimize/_differentialevolution.py

```diff
@@ -1597,40 +1613,49 @@ def _ensure_constraint(self, trial):
 
     def _mutate(self, candidate):
         """Create a trial vector based on a mutation strategy."""
+        
         trial = np.copy(self.population[candidate])
-
         rng = self.random_number_generator
-
-        fill_point = rng.choice(self.parameter_count)
-
-        if self.strategy in ['currenttobest1exp', 'currenttobest1bin']:
-            bprime = self.mutation_func(candidate,
-                                        self._select_samples(candidate, 5))
-        else:
-            bprime = self.mutation_func(self._select_samples(candidate, 5))
-
-        if self.strategy in self._binomial:
-            crossovers = rng.uniform(size=self.parameter_count)
-            crossovers = crossovers < self.cross_over_probability
-            # the last one is always from the bprime vector for binomial
-            # If you fill in modulo with a loop you have to set the last one to
-            # true. If you don't use a loop then you can have any random entry
-            # be True.
-            crossovers[fill_point] = True
-            trial = np.where(crossovers, bprime, trial)
-            return trial
-
-        elif self.strategy in self._exponential:
-            i = 0
-            crossovers = rng.uniform(size=self.parameter_count)
-            crossovers = crossovers < self.cross_over_probability
-            crossovers[0] = True
-            while (i < self.parameter_count and crossovers[i]):
-                trial[fill_point] = bprime[fill_point]
-                fill_point = (fill_point + 1) % self.parameter_count
-                i += 1
-
+    
+        if self.strategy_func:
```

Comment:
Okay, I'm kind of lost in all the changes. Is this something you've changed and made a request for?  I'll gladly accept the change.  But if I've committed like half the changes you made but then you've not made requests for the other half, and they affect things like indentation and duplication, this is going to get very confusing.  :(

Reply:
- Remove lines 1634 to 1641, the lines you added.
- undelete the large bit in red above this section (line numbers 1600-1632)
- I added lines 1619 to 1628, move that to the stat of the method.

(If you'd made this on a feature branch it would be easier for me to make changes directly on the PR, because github often allows maintainers to do that.)